src/asciidocEngine.ts

```diff
@@ -160,11 +161,26 @@ export class AsciidocEngine {
     const antoraSupport = AntoraSupportManager.getInstance(context.workspaceState)
     const antoraAttributes = await antoraSupport.getAttributes(textDocumentUri)
     const baseDir = AsciidocTextDocument.fromTextDocument(textDocument).getBaseDir()
+    let textDocumentExt = path.extname(textDocumentUri.fsPath)
+    const textDocumentName = path.basename(textDocumentUri.fsPath, textDocumentExt)
+    textDocumentExt = textDocumentExt.startsWith('.') ? textDocumentExt.substring(1) : '';
```

Comment:
I think we should move this code in https://github.com/asciidoctor/asciidoctor-vscode/blob/master/src/asciidocTextDocument.ts

Since `baseDir` is not available in a browser environment (such as https://github.dev) it would be better to handle this case in one location.

Reply:
I could move that, but I tried to test things in the browser environment and was running into issues.
Is it a known problem that the include directive does not work in the browser environment? Without includes working my fix doesn't actually apply. I tried to use https://github.dev can created a quick example of two adocs and one including the other and it printed out the `Unresolved directive` text in the preview.
Extensions in general are new to me, but I could not figure out how to get it to work, even when I set the `base_dir` according to https://docs.asciidoctor.org/asciidoctor.js/latest/spec/include-support-matrix/
When the runtime is a browser the browser compiled version of asciidoctor is used and it uses XMLHttpRequest to try to open the includes and i just can't figure out how to get that to work correctly... seems the VSCode extension documentation wants all file accesses to use `vscode.workspace.fs` which makes sense. We could probably set up our own `inludeProcessor` like you've done with Antora but I really don't know anything about Antora and whether or not that works in the browser environment. I think it would still be possible to do, might even be nice to make our own `includeProcessor` that works no matter the situation?

That's a fair bit beyond what i can do at the moment though, I see yours is creating Opal classes and whatnot and I'm just barely hangin on when it comes to Opal.

Reply:
> Is it a known problem that the include directive does not work in the browser environment?

In theory, it could work if the File System API abstraction provided by VS code can read file in a browser environment.
I don't recall if `include` are currently supported or not in a browser environment.

My point is that we should not execute code that relies on Node.js `path` or `fs`  module since they won't necessarily be available in a browser environment and might throw an error.

For now, you can use `'browser' in process && (process as any).browser === true` to test if we are running in a browser environment and not call `path` or `fs`  module.

I would create a `ComputeIntrinsicAttributesDocumentAttributes` in `asciidocTextDocument.ts` which can return an empty object (or undefined) when running in a browser environment. Does it make sense?